### PR TITLE
feat: introduce registry to initialize framework

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of  https://github.com/karmada-io/karmada/issues/1371

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by kerthcet** [comment]
kindly ping @RainbowMango

**Comment by RainbowMango** [comment]
@kerthcet can you help rebase your code and solve the conflicts?

**Comment by kerthcet** [comment]
rebased.

File: pkg/scheduler/framework/plugins/names/names.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package names
+
+const (
+	// APIInstalled is a plugin that checks if the API(CRD) of the resource is installed in the target cluster.
+	APIInstalled = "APIInstalled"
+	// ClusterAffinity is a plugin that checks if a resource selector matches the cluster label.
+	ClusterAffinity = "ClusterAffinity"
+	// TaintToleration is a plugin that checks if a propagation policy tolerates a cluster's taints.
```
Comment (huone1) [inline pkg/scheduler/framework/plugins/names/names.go:8]: it is better to  put  the plugin name  and the plugin together

Comment (kerthcet) [reply pkg/scheduler/framework/plugins/names/names.go]: revert the changes.

**Comment by huone1** [comment]
please merge commits into one commit

**Comment by kerthcet** [comment]
squashed.

**Review by RainbowMango — APPROVED** [review]
/approve
Leave LGTM to @huone1 to confirm.

**Pull request merged**